cocotb/handle.py

```diff
@@ -225,6 +225,28 @@ def __dir__(self):
 class HierarchyObject(RegionObject):
     """Hierarchy objects are namespace/scope objects."""
 
+    def __get_sub_handle_by_name(self, name):
+        try:
+            return self._sub_handles[name]
```

Comment:
This cache has the ability to store every subhandle, so there is the potential for this to take up at least 2x (if not 3-4x because of overhead from Python and PLI handles) the memory of the entire design. I have projects that I simulate with cocotb that have >1e6 accessible objects, keeping all of them the entire lifetime of the program is a bad idea. Could/should we instead move to `lru_cache`? When we finally tie PLI handle lifetimes to Python lifetimes, this will work nicely.

Reply:
Actually, this isn't a great idea. If the design has deep narrow hierarchies, an LRU cache per parent node won't perform well.